src/Raven.Server/Documents/Handlers/Debugging/DocumentDebugHandler.cs

```diff
@@ -54,5 +58,65 @@ public async Task HugeDocuments()
                 writer.WriteEndObject();
             }
         }
+        
+        [RavenAction("/databases/*/debug/documents/scan-corrupted-ids", "GET", AuthorizationStatus.ValidUser, EndpointType.Read, IsDebugInformationEndpoint = true)]
+        public async Task ScanCorruptedIds()
+        {
+            var startEtag = GetIntValueQueryString("startEtag", required: false) ?? 0;
+            var corruptedCount = GetIntValueQueryString("corruptedCount", required: false) ?? int.MaxValue;
+            
+            var corrupted = new List<string>();
+            long? lastEtag = null;
+            
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, Stream.Null))
+                using (context.OpenReadTransaction())
+                {
+                    var documents = Database.DocumentsStorage.GetDocumentsFrom(context, startEtag, 0, long.MaxValue, DocumentFields.Id | DocumentFields.LowerId | DocumentFields.ChangeVector);
+                    var revisions = Database.DocumentsStorage.RevisionsStorage.GetRevisionsFrom(context, startEtag, long.MaxValue, DocumentFields.Id | DocumentFields.LowerId | DocumentFields.ChangeVector);
+                    foreach (var doc in documents.Concat(revisions))
```

Comment:
this is not by the etag order... you will scan all docs before moving on to revisions... 
better use the replication mechanism here?